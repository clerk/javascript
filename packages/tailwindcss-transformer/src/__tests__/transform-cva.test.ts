import { transform } from '../index';

test('cva', () => {
  const result = transform(
    `const button = cva({
      base: "rounded border font-semibold",
      variants: {
        intent: {
          primary: [
            "bg-blue-500",
            "text-white",
            "border-transparent",
            "hover:bg-blue-600",
          ],
          secondary: "border-gray-400 bg-white text-gray-800 hover:bg-gray-100",
        },
        size: {
          small: "px-2 py-1 text-sm",
          medium: "px-4 py-2 text-base",
        },
      },
      compoundVariants: [
        {
          intent: "primary",
          size: "medium",
          className: "uppercase"
        },
      ],
      defaultVariants: {
        intent: "primary",
        size: "medium",
      },
    });
    
    const Button: React.FC<ButtonProps> = ({ children, intent, size, className }) => {
      return <button className={button({ intent, size, className: ['text-white', className] })}>{children}</button>
    }
    
    const ButtonLink: React.FC<LinkProps> = ({ children, intent, size, className }) => {
      return <a className={button({ intent, size, className })}>{children}</a>
    }`,
    {
      styleCache: new Map(),
    },
  );
  expect(result).toMatchSnapshot();
});
