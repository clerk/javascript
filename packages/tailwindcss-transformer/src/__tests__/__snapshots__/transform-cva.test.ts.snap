// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cva 1`] = `
"const button = cva({
      base: "cl-7e77acd4",
      variants: {
        intent: {
          primary: [
            "cl-44eeb93b",
            "cl-5e9a40ae",
            "cl-e9c0d2f3",
            "cl-d8b3f644",
          ],
          secondary: "cl-6537e500",
        },
        size: {
          small: "cl-716a7c80",
          medium: "cl-3f348faf",
        },
      },
      compoundVariants: [
        {
          intent: "primary",
          size: "medium",
          className: "cl-d2cf63c7"
        },
      ],
      defaultVariants: {
        intent: "primary",
        size: "medium",
      },
    });
    
    const Button: React.FC<ButtonProps> = ({ children, intent, size = 'sm', className }) => {
      return <button className={button({ intent: 'primary', size, className: ["cl-5e9a40ae", className] })}>{children}</button>;
    }
    
    const ButtonLink: React.FC<LinkProps> = ({ children, intent, size, className }) => {
      return <a className={button({ intent, size: 'sm', className })}>{children}</a>
    }"
`;
